submit-web/src/hooks/api/useAccountUsers.ts

```diff
@@ -70,3 +70,60 @@ export const useSaveUserProfile = ({
     },
   });
 };
+
+type EditUserRequest = {
+  role_name: string;
+  package_ids?: number[];
+};
+export const editUserRole = (account_user_id: number, data: EditUserRequest) => {
+  return submitRequest<AccountUserWithRole>({
+    url: `/accounts/user/${account_user_id}/role`,
+    method: "patch",
+    data,
+  });
+};
+
+type UseSaveUserRoleParams = {
+  account_user_id: number;
+  options?: Options;
+};
+export const useSaveUserRole = ({
+  account_user_id,
+  options,
+}: UseSaveUserRoleParams) => {
+  return useMutation({
+    mutationFn: (data: EditUserRequest) => {
+      if (!account_user_id) {
+        throw new Error("Account user id is required");
+      }
+
+      return editUserRole(account_user_id, data);
+    },
+    ...options,
+    onSuccess: () => {
+      if (options?.onSuccess) {
+        options.onSuccess();
+      }
+    },
+    onError: (error: any) => {
+      let errorMessage = "An error occurred while updating the user role.";
+      if (error?.response) {
```

Comment:
these if structures here are too much, if need that let's do it once in a common folder and reuse it. Also see if isAxiosError from 'axios' can help you here. Like 
![image](https://github.com/user-attachments/assets/98775f48-ecf9-4467-b7bd-4cd0e9dd7017)

in UpdateRequestWidget component

Reply:
Changed and used isAxiosError, Thanks.